test/cable/channel_test.rb

```diff
@@ -206,4 +208,71 @@ class Futurism::ChannelTest < ActionCable::Channel::TestCase
       assert_mock mock_renderer
     end
   end
+
+  test "renders error message when rendering invalid partial error" do
+    fragment = Nokogiri::HTML.fragment(futurize(partial: "INVALID/PARTIAL", extends: :div) {})
+    signed_params = fragment.children.first["data-signed-params"]
+    subscribe(channel: "Futurism::Channel")
+
+    assert_cable_ready_operation_on("Futurism::Channel:1", operation: "outerHtml",
+                                                           selector: "[data-signed-params='#{signed_params}']",
+                                                           html: /Missing partial INVALID\/_PARTIAL/) do
+      perform :receive, {"signed_params" => [signed_params]}
+    end
+  end
+
+  test "renders error message when wrong variable name" do
+    Post.create title: "Lorem"
+    fragment = Nokogiri::HTML.fragment(futurize(partial: "posts/card", collection: Post.all, as: :wrong_variable_name, extends: :div) {})
+    signed_params = fragment.children.first["data-signed-params"]
+    subscribe(channel: "Futurism::Channel")
+
+    assert_cable_ready_operation_on("Futurism::Channel:1", operation: "outerHtml",
+                                                           selector: "[data-signed-params='#{signed_params}']",
+                                                           html: /undefined local variable or method/) do
+      perform :receive, {"signed_params" => [signed_params]}
+    end
+  end
+
+  def assert_cable_ready_operation_on(stream, operation:, selector:, html:, &block)
```

Comment:
Do you have a link to the respective docs and/or further explanation for me?

TBH it's hard to grok what's going on with `broadcasts` and `clear_messages` etc., and it's hard to see why this preferable to the original, simpler `assert_broadcast_on`

Reply:
Totally get the confusion -- I honestly don't know what it's mostly doing. This is a copy-paste job from https://github.com/rails/rails/blob/5aaaa1630ae9a71b3c3ecc4dc46074d678c08d67/actioncable/lib/action_cable/test_helper.rb#L97

With then a slight modification to build the cable-ready payload. If this style of thing works, I think extracting this to its own file (happy to do so now), and ideally extract to a gem or really just including in https://github.com/hopsoft/cable_ready might be warranted. The API would need to be cleaned up to support multiple message broadcasts and such, but could be helpful to the community.

Reply:
Ah okay, now I get it. Yep, that makes sense. I do think that it would make sense to have this in CR, e.g. to help out here: https://github.com/hopsoft/stimulus_reflex/blob/master/test/broadcasters/page_broadcaster_test.rb#L41

Ping @leastbad @marcoroth @hopsoft

Reply:
I am no testing expert, but I can say that very recently I found myself copy-pasting the `assert_broadcast_on` method into the test class (temporarily!) so that I could redefine the method with a puts in there to spit out what the assert was expecting so I could tweak the data structure hard-coded into the test and make it pass. eg the functionality was working but the test was off by "this much"

I would love to see CR redefine `assert_broadcast_on` to **also** print out the value it's expecting.